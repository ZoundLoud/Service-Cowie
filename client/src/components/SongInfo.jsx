import React from 'react';

const SongInfo = (props) => {
  return (
    <div className="Info-Container">
      <div className="Artist-Name">
      hello
      </div>
      <span className="Song-Name">
      this is the name of the song and this is a test to make sure my text wraps
      </span>
    </div>
  );
};

export default SongInfo;
