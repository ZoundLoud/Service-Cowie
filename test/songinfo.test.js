import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow, render } from 'enzyme';
import SongInfo from '../client/src/components/SongInfo';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('SongInfo Component', () => {
  const props = {
    artistname: 'Etha Crona Sr.',
    songname: 'Qui blanditiis quas.',
  };
  it('should render SongInfo without throwing an error', () => {
    const song = shallow(<SongInfo song={props.artistname} />);
    expect(toJson(song)).toMatchSnapshot();
  });
  it('should render correctly in "debug" mode', () => {
    const debug = shallow(<SongInfo song={props.artistname} debug />);
    expect(toJson(debug)).toMatchSnapshot();
  });
  it('should mount correctly', () => {
    const song = mount(<SongInfo song={props.artistname} />);
    expect(toJson(song)).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const song = render(<SongInfo song={props.artistname} />);
    expect(toJson(song)).toMatchSnapshot();
  });
});
