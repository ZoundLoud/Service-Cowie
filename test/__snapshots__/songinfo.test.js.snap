// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongInfo Component should render SongInfo without throwing an error 1`] = `
<div
  className="Info-Container"
>
  <div
    className="Artist-Name"
  />
  <span
    className="Song-Name"
  />
</div>
`;

exports[`SongInfo Component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
